src/Microsoft.OData.Client/ODataAnnotatableExtensions.cs

```diff
@@ -30,12 +30,12 @@ public static T GetAnnotation<T>(this ODataAnnotatable annotatable)
 
         private static class InternalDictionary<T> where T : class
         {
-            private static readonly ConcurrentDictionary<ODataAnnotatable, T> Dictionary =
-                new ConcurrentDictionary<ODataAnnotatable, T>();
+            private static readonly ConditionalWeakTable<ODataAnnotatable, T> Dictionary =
+                new ConditionalWeakTable<ODataAnnotatable, T>();
 
             public static void SetAnnotation(ODataAnnotatable annotatable, T annotation)

```

Comment:
>SetAnnotation [](start = 31, length = 13)

Change looks reasonable.

Since we are changing semantics on this internal class, I wonder if we should rename this to "AddAnnotation" (and perhaps add an assert that the table doesn't already contain the specified annotatable?)